src/python/pants/option/custom_types.py

```diff
@@ -53,14 +54,33 @@ def list_option(s):
   return ListValueComponent.create(s)
 
 
+# Match any repeated instances of the directory separator, unless they occur at the start. This
+# will normalize file and target paths as required by the engine, without stripping a leading // in
+# target addresses.
+_repeated_separator_pattern = re.compile('(?!^){}+'.format(re.escape(os.sep)))
+
+
+def _normalize_directory_separators(s):
+  """Coalesce runs of consecutive instances of `os.sep` in `s`, e.g. '//' -> '/' on POSIX.
+
+  The engine will use paths or target addresses either to form globs or to string-match against, and
+  including the directory separator '/' multiple times in a row e.g. '//' produces an equivalent
+  glob as with a single '/', but produces a different actual string, which will cause the engine to
+  fail to glob file paths or target specs correctly.
+
+  TODO: give the engine more control over matching paths so we don't have to sanitize the input!
+  """
+  return re.sub(_repeated_separator_pattern, os.sep, s)
```

Comment:
Is this intentionally not using `os.path.normpath` due to the warning in https://docs.python.org/2/library/os.path.html#os.path.normpath about symlinks? ... I don't really know whether that is something we should be concerned about with regard to options values.

Reply:
No, it should probably just use `os.path.normpath()`! Target options probably shouldn't be modified at all either, since they're not the ones causing this issue -- will fix.

Reply:
Done!